### PR TITLE
Search by OrderId and Address account

### LINKED ISSUE #691: Search by address
**Description**

Right now, search only supports searching Order ids across different networks.

This issue is about searching by address as well, for the upcoming Address details page

**Proposal**

One simple way of supporting both proposed by Anxo is to match the shape of the input:
1. If it matches `0x[a-fA-F0-9]{112}`; use the current orderId search
2. If it matches `0x[a-fA-F0-9]{40}`; load the address page for the respective network
3. Otherwise, show the not found page

**Alternatives**

No other ideas for now but open to suggestions

**Additional context**

I have not tested the regexes

### PR DESCRIPTION
**Pull request by henrypalacios**
# Summary

Closes #691 

**Proposal:**
- [x] Allow search by `OrderId` and `Address Account`
- [x] Otherwise, show the **not found page**

## To Test
1. Go to explorer [search](https://pr752--gpui.review.gnosisdev.com/)
2.1 Search by **Address account** -> i.e: `0x04a66cbba0485d7b21af836f52b711401300fddb` 
2.2 Search by **Order Id** -> i.e: `0x88537c74a03c324464d6c269d99b29e7289b105766c5368f77c76c241e1d5da8b6bad41ae76a11d10f7b0e664c5007b908bc77c9615b2dac`

### CODE REVIEW

**Comment by elena-zh** [comment]
Changes LGTM: I'm able to search by an order ID  and an address (full match).
But an address info is available in the Mainnet only, even if I search for it when the explorer is in XDAI network 
https://watch.screencastify.com/v/UvUk0c1NJENLRponEaYG

I hope, it will be fixed in #747

**Comment by henrypalacios** [comment]
Thanks @elena-zh  for the feedback, Then I think it is best to keep the URL `network prefix` in the search.
Let me try!

File: src/components/orders/OrderNotFound/index.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { useParams } from 'react-router'
 import styled from 'styled-components'
-import { Search } from 'components/orders/Search'
+import { Search } from 'apps/explorer/components/common/Search'
```
Comment (alongoni) [inline src/components/orders/OrderNotFound/index.tsx:4]: With this change, we might be missing some CSS. 
image
The search box should be aligned to the left. 

image

**Review by alongoni — APPROVED** [review]
Looks and works fine! 

(Only one comment regarding search box on order not found page)

File: src/apps/explorer/components/common/Search/Search.styled.ts [hunk 2]
```diff
@@ -1,11 +1,18 @@
-import styled from 'styled-components'
+import styled, { css, FlattenSimpleInterpolation } from 'styled-components'
 import SVG from 'react-inlinesvg'
 
 export const Wrapper = styled.form`
   display: flex;
-  width: 100%;
-  max-width: 50rem;
-  margin: 0 auto;
+  ${({ className }): FlattenSimpleInterpolation =>
```
Comment (henrypalacios) [inline src/apps/explorer/components/common/Search/Search.styled.ts:6]: @alongoni [`orders/Search`](https://github.com/gnosis/gp-ui/blob/691-search-by-address/src/components/orders/Search/index.tsx#L8) and [`explorer/common/Search`](https://github.com/gnosis/gp-ui/blob/691-search-by-address/src/apps/explorer/components/common/Search/index.tsx#L8) look identical, except for a new lines of CSS, 
I've added the difference through a `className`.

[`src/components/orders/Search`](https://github.com/gnosis/gp-ui/blob/691-search-by-address/src/components/orders/Search/index.tsx#L8) would be deprecated, @alfetopito we could delete it?

File: test/utils/miscellaneous.spec.ts [hunk 3]
```diff
@@ -89,3 +90,59 @@ describe('getToken', () => {
...
+  it('should be an orderId', () => {
+    const text =
+      '0x405bd0278c11399f84f10e19fb9b45123996e7d0a68a60ddebc3b9581576b484ff714b8b0e2700303ec912bd40496c3997ceea2b614b17d9'
+
+    const result = isAnOrderId(text)
+
+    expect(result).toBe(true)
+  })
+
+  it('not be an orderId', () => {
```
Comment (anxolin) [inline test/utils/miscellaneous.spec.ts:104]: uber-nitpicking: The name of the `it` is not completly coherent with the other one. 

This one should be according to the other `shouldn't be an order id`, however, I would simplify all of them to
`Is orderId`, `Is not orderId`

File: test/utils/miscellaneous.spec.ts [hunk 4]
```diff
@@ -89,3 +90,59 @@ describe('getToken', () => {
     })
   })
 })
+
+describe('isAnOrderId', () => {
```
Comment (anxolin) [inline test/utils/miscellaneous.spec.ts:94]: I'm very glad to see you did some tests for this

File: test/hooks/useSearchSubmit.test.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+function wrapperMemoryRouter(props: Props): JSX.Element {
+  return (
+    <>
+      <Router history={props.history}>{props.children}</Router>
+    </>
+  )
+}
+
+describe('useSearchSubmit', () => {
```
Comment (anxolin) [inline test/hooks/useSearchSubmit.test.tsx:21]: Awesome you did a test for this!

**Comment by elena-zh** [comment]
Changes LGTM. 
A tiny issue: to start 'order' word with the capital letter in the placeholder
order with capital

**Comment by alfetopito** [comment]
> @anxolin I think the approach is great!
> 
> > Not for this release, we really need it out, but maybe we can hint somehow the network the user is looking for orders.
> > ...
> > Look in the current network
> > If no results, try to find in another network
> > if there's results, forward the user there
> 
> @alfetopito can I create a PR for this?

Yes, please!

This is actually one of the follow ups I mentioned. Very good to have but not a blocker for the release

**Pull request merged**